### ISSUE TITLE
GPEL3101A programming docs ?

### ISSUE THREAD

**Issue by Retsim** [kaero/t-on#2]
Hello,

Thanks for this repository !
I successfully dumped my SPI firmware, and I checked using this viewer if it looked OK (it was).
I even flashed your others firmwares to try.
Basically, I switched from meets (jap) firmware to ON firmware, and now playing in english !

We are interesting into looking at reversing more this tama, and we look for the GPEL3101A programming guide from GeneralPlus.
How did you find this first PDF ?
Any chance you have the others ?

Thanks !

**Comment by ianling** [comment]
I have reached out to GeneralPlus and tried asking politely for the programming docs, but they would not give them to me. Outside of those docs being leaked somehow, I'm not sure we will ever see them.

**Comment by kaero** [comment]
Hi!

I tried to reach out via email GP and some resellers who still have this SoC in stocks, but had no luck to put on hands the full datasheet :(

The preliminary datasheet was googled using some trivial requests like "GPEL31xx datasheet" :)

Also some chinese web sites (for example [this one](http://www.rf-china.com/n42/gpl.asp)) mentions the programming guide for this SoC, but have no idea where to find it.

Anyway, the preliminary datasheet is a good jumpstart for reversing. I had dumped the first 7 seconds of SPI reads during the boot to have clues where to place the trampoline code. I'm trying to figure out memory map experimentally to dump the SoC ROM by reading it in runtime and dumping to SPI flash. 

I'd [modded the t-on board placing the socket for flash](https://imgur.com/dt5NkbK) for easy experiments with the flash.

Previously I tried to keep this repo up to date with ready to use results, but if you interested in, i can just share all i have on hands. I think, we can setup some chat or mail-list if you are interested in reversing the tamagotchi-on. I'll be pleased to help and get helped :) My progress is slow because this reversing project have the lowest priority of my work & hobby projects, so i do it time to time.
